api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2555,29 +2591,104 @@ public void composeObject(
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, ServerException,
           XmlParserException {
-    if ((bucketName == null) || (bucketName.isEmpty())) {
-      throw new IllegalArgumentException("bucket name cannot be empty");
-    }
 
-    checkObjectName(objectName);
+    ComposeObjectArgs.Builder builder =
+        ComposeObjectArgs.builder()
+            .bucket(bucketName)
+            .object(objectName)
+            .extraHeaders(Multimaps.forMap(headerMap))
+            .sources(sources)
+            .sse(sse);
 
-    if (sources.isEmpty()) {
-      throw new IllegalArgumentException("compose sources cannot be empty");
-    }
+    composeObject(builder.build());
+  }
 
-    checkWriteRequestSse(sse);
+  /**
+   * Creates an object by combining data from different source objects using server-side copy.
+   *
+   * <pre>Example:{@code
+   * List<ComposeSource> sourceObjectList = new ArrayList<ComposeSource>();
+   *
+   * sourceObjectList.add(
+   *    ComposeSource.builder().bucket("my-job-bucket").object("my-objectname-part-one").build());
+   * sourceObjectList.add(
+   *    ComposeSource.builder().bucket("my-job-bucket").object("my-objectname-part-two").build());
+   * sourceObjectList.add(
+   *    ComposeSource.builder().bucket("my-job-bucket").object("my-objectname-part-three").build());
+   *
+   * // Create my-bucketname/my-objectname by combining source object list.
+   * minioClient.composeObject(
+   *    ComposeObjectArgs.builder()
+   *        .bucket("my-bucketname")
+   *        .object("my-objectname")
+   *        .sources(sourceObjectList)
+   *        .build());
+   *
+   * // Create my-bucketname/my-objectname with user metadata by combining source object
+   * // list.
+   * minioClient.composeObject(
+   *     ComposeObjectArgs.builder()
+   *        .bucket("my-bucketname")
+   *        .object("my-objectname")
+   *        .sources(sourceObjectList)
+   *        .extraHeaders(Multimaps.forMap(userMetadata))
+   *        .build());
+   *
+   * // Create my-bucketname/my-objectname with user metadata and server-side encryption
+   * // by combining source object list.
+   * minioClient.composeObject(
+   *   ComposeObjectArgs.builder()
+   *        .bucket("my-bucketname")
+   *        .object("my-objectname")
+   *        .sources(sourceObjectList)
+   *        .extraHeaders(Multimaps.forMap(userMetadata))
```

Comment:
`ObjectWriteArgs` comes with `headers` and `userMetadata`. You would need to avoid `extraHeaders` and `extraQueryParams` usage unless required in examples and references.